enterprise/frontend/src/embedding-sdk/lib/version-utils.unit.spec.ts

```diff
@@ -0,0 +1,96 @@
+import { isSdkVersionCompatibleWithMetabaseVersion } from "./version-utils";
+
+jest.mock("../config", () => ({
+  getEmbeddingSdkVersion: jest.fn().mockReturnValue("1.0.0"),
+}));
+
+const expectCompatibility = ({
+  mbVersion,
+  sdkVersion,
+  expected,
+}: {
+  mbVersion: string;
+  sdkVersion: string;
+  expected: boolean;
+}) => {
+  it(`expect sdk version ${sdkVersion} and mb version ${mbVersion} to be ${expected ? "compatible" : "incompatible"}`, () => {
+    expect(
+      isSdkVersionCompatibleWithMetabaseVersion({
+        mbVersion,
+        sdkVersion,
+      }),
+    ).toBe(expected);
+  });
+};
+
+describe('sdk version utils, naming used: "{0,1}.{major}.{minor}"', () => {
+  describe('should return true only if the "major" version is the same', () => {
+    expectCompatibility({
+      mbVersion: "v0.52.10",
+      sdkVersion: "0.52.10",
+      expected: true,
+    });
+    expectCompatibility({
+      mbVersion: "v1.50.10",
+      sdkVersion: "0.51.10",
+      expected: false,
+    });
+    expectCompatibility({
+      mbVersion: "v1.50.10",
+      sdkVersion: "0.53.10",
+      expected: false,
+    });
+  });
+
+  describe('should ignore "minors"', () => {
+    expectCompatibility({
+      mbVersion: "v1.50.10",
+      sdkVersion: "0.50.11",
+      expected: true,
+    });
+
+    expectCompatibility({
+      mbVersion: "v1.50.10",
+      sdkVersion: "0.50.9",
+      expected: true,
+    });
+  });
+
+  describe("sdk version 0.xx.yy should be compatible both with MB 0.xx.yy (OSS) and 1.xx.yy (EE)", () => {
```

Comment:
Why do we say the SDK is compatible with OSS version? Our doc only says to [use EE jar](https://www.metabase.com/docs/latest/embedding/sdk/quickstart). I don't think we mention OSS anywhere.

Reply:
I just don't want us to shoot ourselves in the foot later on. What if we just recommend users to always use the EE jar, they could keep using the OSS features couldn't they? Since they don't have a token. EE binary will just work like an OSS one.

Reply:
Solved this on slack: https://metaboat.slack.com/archives/C063Q3F1HPF/p1731678809265209?thread_ts=1731678227.620839&cid=C063Q3F1HPF